unit/src/DependencyResolverTest.C

```diff
@@ -144,3 +144,21 @@ TEST_F(DependencyResolverTest, dependsOnTest)
   EXPECT_FALSE(_tree.dependsOn("m1", "something_else"));
   EXPECT_FALSE(_tree.dependsOn("something_else", "k0"));
 }
+
+TEST_F(DependencyResolverTest, cyclicTest)
+{
+  try
+  {
+    // Attempt to insert a dependency that results in cyclicity
+    _resolver.insertDependency("a", "b");
+    FAIL() << "missing expected exception";
+  }
+  catch (const std::exception & e)
```

Comment:
You can then optionally call this method:
https://github.com/idaholab/moose/blob/devel/framework/include/utils/DependencyResolver.h#L157, to print out a detailed error message which is super fancy, like deserves a "Holy Dooley" kind of fancy.